src/OpenLoco/src/Effects/VehicleCrashEffect.cpp

```diff
@@ -1,16 +1,84 @@
 #include "VehicleCrashEffect.h"
+#include "Audio/Audio.h"
+#include "Entities/EntityManager.h"
+#include "Map/TileManager.h"
+#include "SplashEffect.h"
 
 #include <OpenLoco/Interop/Interop.hpp>
 
 using namespace OpenLoco::Interop;
 
 namespace OpenLoco
 {
+    // There are a total of 12 sprites for this effect, the paint code divides this by 256.
+    static constexpr uint16_t kMaxSplashFrames = 12 * 256;
+
+    // Advances the frame counter by 85 each tick.
+    static constexpr uint16_t kAnimationSpeed = 85;
+
+    // Unknown which unit this is in, but it is used to calculate the acceleration for gravity.
+    static constexpr int32_t kGravity = 5041;
+
     // 0x004406A0
     void VehicleCrashParticle::update()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x004406A0, regs);
+        invalidateSprite();
```

Comment:
in vanilla this was invalidate full rather than eighth. But don't think we really care.

Reply:
You are right, I didn't look too close into that function was I thought its most likely doing full given the lack of parameter for the zoom level. Should refactor that in a follow up PR to just have `invalidate()` or `invalidate(ZoomLevel)`, invalidateSprite seems slightly misleading as the sprite is not really affected.